tornado/log.py

```diff
@@ -32,12 +32,19 @@
 
 import logging
 import logging.handlers
+import os
 import sys
 
 from tornado.escape import _unicode
 from tornado.util import unicode_type, basestring_type
 
 try:
+    import colorama
+    colorama.init()
```

Comment:
I'm not really comfortable calling `colorama.init` at import time. It looks like it may not be idempotent, and replacing `sys.std{out,err}` is intrusive enough that there should be a more definite opt-in than "the colorama package is installed" (colorama's issue tracker shows that its approach has quite a few quirks). 

What if we leave the call to `colorama.init()` to the application and replace the `if colorama is not None` calls below with `if sys.stderr is getattr(colorama, 'wrapped_stderr', object())`? That way we'd be able to detect the use of `colorama` as an alternative indicator that ANSI color is supported (instead of `curses`'s inspection of `$TERM` and the `terminfo` database)

It looks like windows 10 also has a native ANSI-aware terminal (see tartley/colorama#104), so another approach we could take would be to ignore colorama and figure out how to detect that terminal in the absence of `curses` and `terminfo`.